src/c/examples/template.c

```diff
@@ -38,7 +38,9 @@ static void dump_attributes (iot_logger_t *lc, devsdk_resource_attr_t attrs)
   iot_data_map_iter ((iot_data_t *)attrs, &iter);
   while (iot_data_map_iter_next (&iter))
   {
-    iot_log_debug (lc, "    %s = %s", iot_data_map_iter_string_key (&iter), iot_data_map_iter_string_value (&iter));
+    char *json = iot_data_to_json (iot_data_map_iter_value (&iter));
```

Comment:
Recommend wrapping in an if such as:

  if (lc && lc->level >= IOT_LOG_DEBUG)
  {
     char *json = iot_data_to_json (iot_data_map_iter_value (&iter));
...

So that the json generation and free is only done if at debug level instead of always...